zebra-network/src/protocol/message.rs

```diff
@@ -307,6 +308,26 @@ pub enum Message {
     MerkleBlock {/* XXX add fields */},
 }
 
+impl<E> From<E> for Message
+where
+    E: Error,
+{
+    fn from(e: E) -> Self {
+        Message::Reject {
+            message: e.to_string(),
+
+            // The generic case, impls for specific error types should
+            // use specific varieties of `RejectReason`.
+            ccode: RejectReason::Other,
+
+            reason: e.source().unwrap().to_string(),
+
+            // Allow this to be overriden but not populated by default, methinks.
```

Comment:
later, this could be populated by the same logic that sets the `ccode` to something other than `Other`.